agent/lib/kontena/workers/node_stats_worker.rb

```diff
@@ -116,13 +125,15 @@ def publish_node_stats
         network: network_traffic,
         time: Time.now.utc.to_s
       }
-      rpc_client.async.notification('/nodes/stats', [data])
-      send_statsd_metrics(data)
+    end
+
+    # @param [Hash] data
+    def send_node_stats(data)
+      rpc_client.async.notification('/nodes/stats', [@node.id, data])
     end
 
     # @param [Hash] event
     def send_statsd_metrics(event)
-      return unless statsd
       key_base = "#{docker_info['Name']}"
```

Comment:
I would very much also like to remove the `Docker.info` in here..

* The statsd `docker_info['Name']` could be changed to use the `@node.name` instead? It looks like ATM nodes with duplicate hostnames would result in overlapping statsd keys, whereas the `@node.name` is unique?

* The `docker_info['DockerRootDir']` is also used for the `filesystem.name` path... but that's inaccurate, because the disk stats are actually from `Vmstat.disk('/')`, which might be a different filesystem?